src/hotspot/share/runtime/thread.hpp

```diff
@@ -1423,6 +1423,8 @@ class JavaThread: public Thread {
  public:
   // Returns the running thread as a JavaThread
   static inline JavaThread* current();
+  // Returns the current thread as a JavaThread, or NULL if not attached
+  static inline JavaThread* current_or_null();
```

Comment:
I hadn't intended to suggest the introduction of this method, but I see now how my comment led to it. It is fine.